src/Module/Contact/Repair.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+namespace Friendica\Module\Contact;
+
+use Friendica\BaseModule;
+use Friendica\Core\Protocol;
+use Friendica\Core\Renderer;
+use Friendica\Core\Session;
+use Friendica\DI;
+use Friendica\Model;
+use Friendica\Module\Contact;
+use Friendica\Network\HTTPException\BadRequestException;
+use Friendica\Network\HTTPException\ForbiddenException;
+use Friendica\Util\Strings;
+
+/**
+ * GUI for repairing contact details
+ */
+class Repair extends BaseModule
+{
+	public static function init(array $parameters = [])
+	{
+		if (!Session::isAuthenticated()) {
+			throw new ForbiddenException(DI::l10n()->t('Permission denied.'));
+		}
+	}
+
+	public static function post(array $parameters = [])
+	{
+		$cid = $parameters['contact'];
+
+		$contact = Model\Contact::selectFirst([], ['id' => $cid, 'uid' => local_user()]);
+		if (empty($contact)) {
+			throw new BadRequestException(DI::l10n()->t('Contact not found.'));
+		}
+
+		$name        = ($_POST['name'] ?? '') ?: $contact['name'];
+		$nick        = $_POST['nick'] ?? '';
+		$url         = $_POST['url'] ?? '';
+		$alias       = $_POST['alias'] ?? '';
+		$request     = $_POST['request'] ?? '';
+		$confirm     = $_POST['confirm'] ?? '';
+		$notify      = $_POST['notify'] ?? '';
+		$poll        = $_POST['poll'] ?? '';
+		$attag       = $_POST['attag'] ?? '';
+		$photo       = $_POST['photo'] ?? '';
+		$remote_self = $_POST['remote_self'] ?? false;
+		$nurl        = Strings::normaliseLink($url);
+
+		$r = DI::dba()->update(
+			'contact',
+			[
+				'name'        => $name,
+				'nick'        => $nick,
+				'url'         => $url,
+				'nurl'        => $nurl,
+				'alias'       => $alias,
+				'request'     => $request,
+				'confirm'     => $confirm,
+				'notify'      => $notify,
+				'poll'        => $poll,
+				'attag'       => $attag,
+				'remote_self' => $remote_self,
+			],
+			['id' => $contact['id'], 'uid' => local_user()]
+		);
+
+		if ($photo) {
+			DI::logger()->notice('Updating photo.', ['photo' => $photo]);
+
+			Model\Contact::updateAvatar($photo, local_user(), $contact['id']);
+		}
+
+		if ($r) {
+			info(DI::l10n()->t('Contact settings applied.') . EOL);
+		} else {
+			notice(DI::l10n()->t('Contact update failed.') . EOL);
+		}
+
+		return;
+	}
+
+	public static function content(array $parameters = [])
+	{
+		$cid = $parameters['contact'];
+
+		$cid = $parameters['contact'];
+
+		$contact = Model\Contact::selectFirst([], ['id' => $cid, 'uid' => local_user()]);
+		if (empty($contact)) {
+			throw new BadRequestException(DI::l10n()->t('Contact not found.'));
+		}
+
+		if (empty(DI::page()['aside'])) {
```

Comment:
It is initialized on line 42 of `src/App/Page.php` and therefore this initialization isn't necessary.